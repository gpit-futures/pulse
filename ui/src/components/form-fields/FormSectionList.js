import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class FormSectionList extends PureComponent {
  static propTypes = {
    title: PropTypes.string.isRequired,
  };

  render() {
    const { title, children } = this.props;

    return (
      <div className="form-group-section-list">
        <div className="form-group-section-heading">
          <label className="control-label">{title}</label>
        </div>
        {children}
      </div>
    )
  }
}

