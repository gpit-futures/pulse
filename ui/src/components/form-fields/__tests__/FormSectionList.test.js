import React from 'react';
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-15';

configure({ adapter: new Adapter() });

import FormSectionList from '../FormSectionList';

const title = 'Test title';

describe('Component <FormSectionList />', () => {
  it('should renders with all props correctly', () => {
    const component = shallow(
      <FormSectionList
        title={title}
      >
        <span className="FormSectionList-children">FormSectionList children</span>
      </FormSectionList>);

    expect(component.find('.form-group-section-list')).toHaveLength(1);
    expect(component.find('.form-group-section-heading')).toHaveLength(1);
    expect(component.find('.control-label')).toHaveLength(1);
    expect(component.find('.control-label').text()).toEqual(title);
    expect(component.find('.FormSectionList-children')).toHaveLength(1);
    expect(component.instance().props.title).toEqual(title);

    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
