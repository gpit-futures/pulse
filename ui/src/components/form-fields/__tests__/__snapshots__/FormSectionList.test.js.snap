// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <FormSectionList /> should renders with all props correctly 1`] = `
<div
  className="form-group-section-list"
>
  <div
    className="form-group-section-heading"
  >
    <label
      className="control-label"
    >
      Test title
    </label>
  </div>
  <span
    className="FormSectionList-children"
  >
    FormSectionList children
  </span>
</div>
`;
